opennms-correlation/drools-correlation-engine/src/test/opennms-home/etc/drools-engine.d/persistState/PersistState.drl

```diff
@@ -4,13 +4,22 @@ import org.opennms.netmgt.correlation.drools.DroolsCorrelationEngine;
 import org.opennms.netmgt.xml.event.Event;
 import org.opennms.netmgt.model.events.EventBuilder;
 import org.drools.core.spi.KnowledgeHelper;
+import org.drools.core.QueryResultsImpl;
 
 global DroolsCorrelationEngine engine;
 
 declare Thing
 	name: String
 end
 
+declare PersistedTestFact
+  factId: String
+end
+
+query "find PersistedTestFact by factId" (String searchFactId)
```

Comment:
query, nice

Reply:
This is all taken from @UberPinguin 's PR :-)